src/FunderMaps.Core/Notification/NotifyContext.cs

```diff
@@ -0,0 +1,52 @@
+namespace FunderMaps.Core.Notification
+{
+    /// <summary>
+    ///     Notification handler status.
+    /// </summary>
+    public enum NotifyContextStatus
+    {
+        /// <summary>
+        ///     Notify was submitted to handler.
+        /// </summary>
+        Submitted,
+
+        /// <summary>
+        ///     Notification was handled with success.
+        /// </summary>
+        Success,
+
+        /// <summary>
+        ///     Notification was handled with fail.
+        /// </summary>
+        Failed,
+
+        /// <summary>
+        ///     Notification was not handled.
+        /// </summary>
+        NotHandled
+    }
+
+    /// <summary>
+    ///     Notification context.
+    /// </summary>
+    public class NotifyContext
+    {
+        /// <summary>
+        ///     Envelope.
+        /// </summary>
+        public Envelope Envelope { get; set; }
+
+        /// <summary>
+        ///     Handler status.
+        /// </summary>
+        public NotifyContextStatus Status { get; private set; } = NotifyContextStatus.Submitted;
+
+        /// <summary>
+        ///     Set handler status.
+        /// </summary>
+        public void SetStatus(NotifyContextStatus status)
```

Comment:
What is the benefit of this assignment structure?

Reply:
The context is being passed around after the operation has completed. We want to keep a reference to the context so it can be manipulated

Reply:
Maybe there is some time for an example later